query-engine/core/Cargo.toml

```diff
@@ -34,3 +37,6 @@ schema = { path = "../schema" }
 lru = "0.7.7"
 enumflags2 = "0.7"
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+pin-project = "1"
```

Comment:
This enables [pin projection](https://doc.rust-lang.org/std/pin/index.html#projections-and-structural-pinning), and is used to spawn tasks in a Wasm-compatible fashion.

Reply:
Out of curiosity. Why is it so that Wasm requires memory pinning?  what is what it's pinning? is it internals to memory addressess of .TEXT or .DATA sections?

Reply:
See: https://github.com/prisma/prisma-engines/pull/4445#discussion_r1397465201.